src/ddoc_cache/src/ddoc_cache_tables.erl

```diff
@@ -0,0 +1,69 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(ddoc_cache_tables).
+-behaviour(gen_server).
+-vsn(1).
+
+
+-export([
+    start_link/0
+]).
+
+-export([
+    init/1,
+    terminate/2,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    code_change/3
+]).
+
+
+-include("ddoc_cache.hrl").
+
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+
+init(_) ->
+    BaseOpts = [public, named_table],
```

Comment:
`Certainly can't have the LRU updates not single-threaded as we can't do multi-key atomic compare and replace type operations.` Is that actually needed? I think the duplicate bag approach above might allow us to skip that entirely. We would still need to figure out what to do for the `?LRU` entries in that if we used `os:timestamp()` we could potentially end up with duplicate entries, and I don't think there's an `ordered_duplicate_bag` ETS table. But I think there's potential for a less strict LRU that accomplishes the same things.